spec/api/work_package_form_resource_spec.rb

```diff
@@ -0,0 +1,162 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'rack/test'
+
+describe 'API v3 Work package form resource', type: :request do
+  include Rack::Test::Methods
+  include Capybara::RSpecMatchers
+
+  let(:project) { FactoryGirl.create(:project, is_public: false) }
+  let(:work_package) { FactoryGirl.create(:work_package, project: project) }
+  let(:authorized_user) { FactoryGirl.create(:user, member_in_project: project) }
+  let(:unauthorized_user) { FactoryGirl.create(:user) }
+
+  describe '#post' do
+    shared_examples_for 'valid payload' do
+      subject { response.body }
+
+      it { expect(response.status).to eq(200) }
+
+      it { is_expected.to have_json_path('_embedded/payload') }
+
+      it { is_expected.to have_json_path('_embedded/payload/lockVersion') }
+
+      it { is_expected.to have_json_path('_embedded/payload/subject') }
+
+      it_behaves_like 'API V3 formattable', '_embedded/payload/description' do
+        let(:format) { 'textile' }
+        let(:raw) { defined?(raw_value) ? raw_value : work_package.description.to_s }
+        let(:html) {
+          defined?(html_value) ? html_value : ('<p>' + work_package.description.to_s + '</p>')
+        }
+      end
+    end
+
+    shared_examples_for 'having no errors' do
+      it {
+        expect(subject.body).to be_json_eql({}.to_json)
+                                    .at_path('_embedded/validationErrors')
+      }
+    end
+
+    shared_examples_for 'having an error' do |property|
+      it { expect(subject.body).to have_json_path("_embedded/validationErrors/#{property}") }
+
+      describe 'error body' do
+        let(:error_id) { 'urn:openproject-org:api:v3:errors:PropertyConstraintViolation' }
+
+        let(:error_body) {
+          parse_json(subject.body)['_embedded']['validationErrors'][property]
+        }
+
+        it { expect(error_body['errorIdentifier']).to eq(error_id) }
+      end
+    end
+
+    let(:post_path) { "/api/v3/work_packages/#{work_package.id}/form" }
+    let(:valid_params) do
+      {
+          _type: 'WorkPackage',
+          lockVersion: work_package.lock_version
+      }
+    end
+
+    subject(:response) { last_response }
+
+    shared_context 'post request' do
+      before(:each) do
+        allow(User).to receive(:current).and_return current_user
+        post post_path, (params ? params.to_json : nil), 'CONTENT_TYPE' => 'application/json'
+      end
+    end
+
+    let(:params) {}
+    let(:current_user) { authorized_user }
+
+    let(:valid_params) do
+      {
+          _type: 'WorkPackage',
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.